src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -0,0 +1,972 @@
+    }
+
+    // If from address is a taddr, select UTXOs to spend
+    CAmount selectedUTXOAmount = 0;
+    bool selectedUTXOCoinbase = false;
+    if (isfromtaddr_) {
+        std::vector<SendManyInputUTXO> selectedTInputs;
+        for (SendManyInputUTXO & t : t_inputs_) {
+            bool b = std::get<3>(t);
+            if (b) {
+                selectedUTXOCoinbase = true;
+            }
+            selectedUTXOAmount += std::get<2>(t);
+            selectedTInputs.push_back(t);
+            if (selectedUTXOAmount >= targetAmount) {
+                break;
+            }
+        }
+        t_inputs_ = selectedTInputs;
+        t_inputs_total = selectedUTXOAmount;
+
+        // update the transaction with these inputs
+        CMutableTransaction rawTx(tx_);
+        for (SendManyInputUTXO & t : t_inputs_) {
+            uint256 txid = std::get<0>(t);
+            int vout = std::get<1>(t);
+            CAmount amount = std::get<2>(t);
+            CTxIn in(COutPoint(txid, vout));
+            rawTx.vin.push_back(in);
+        }
+        tx_ = CTransaction(rawTx);
+    }
+
+    LogPrint("asyncrpc", "%s: spending %s to send %s with fee %s\n",
+            getId().substr(0,10), FormatMoney(targetAmount, false), FormatMoney(sendAmount, false), FormatMoney(minersFee, false));
+    LogPrint("asyncrpc", " -  transparent input: %s (to choose from)\n", FormatMoney(t_inputs_total, false));
+    LogPrint("asyncrpc", " -      private input: %s (to choose from)\n", FormatMoney(z_inputs_total, false));
+    LogPrint("asyncrpc", " - transparent output: %s\n", FormatMoney(t_outputs_total, false));
+    LogPrint("asyncrpc", " -     private output: %s\n", FormatMoney(z_outputs_total, false));
+    LogPrint("asyncrpc", " -                fee: %s\n", FormatMoney(minersFee, false));
+
+    /**
+     * SCENARIO #1
+     * 
+     * taddr -> taddrs
+     * 
+     * There are no zaddrs or joinsplits involved.
+     */
+    if (isPureTaddrOnlyTx) {
+        add_taddr_outputs_to_tx();
+        
+        CAmount funds = selectedUTXOAmount;
+        CAmount fundsSpent = t_outputs_total + minersFee;
+        CAmount change = funds - fundsSpent;
+        
+        if (change > 0) {
+            add_taddr_change_output_to_tx(change);
+
+            LogPrint("asyncrpc", "%s: transparent change in transaction output (amount=%s)\n",
+                    getId().substr(0, 10),
+                    FormatMoney(change, false)
+                    );
+        }
+        
+        Object obj;
+        obj.push_back(Pair("rawtxn", EncodeHexTx(tx_)));
+        sign_send_raw_transaction(obj);
+        return true;
+    }
+    /**
+     * END SCENARIO #1
+     */
+
+    
+    // Prepare raw transaction to handle JoinSplits
+    CMutableTransaction mtx(tx_);
+    mtx.nVersion = 2;
+    crypto_sign_keypair(joinSplitPubKey_.begin(), joinSplitPrivKey_);
+    mtx.joinSplitPubKey = joinSplitPubKey_;
+    tx_ = CTransaction(mtx);
```

Comment:
We shouldn't be duplicating this every time we construct a transaction with JoinSplits; it should be abstracted out somewhere (and similarly for adding the signature). It's sufficient to file a ticket to track this.